### PR TITLE
Generate swagger as yaml

### PR DESCRIPTION
**Pull request by ravidbro**

### CODE REVIEW

File: hammock/swaggerize.py [hunk 1]
```diff
@@ -39,14 +37,14 @@ def _generate_param_dict(route, name, arg, where, required):
 
 
 def generate_swagger(package):
-    doc = OrderedDict(
```
Comment (posener) [inline hammock/swaggerize.py]: weren't there a reason for the ordered dict?

File: hammock/swaggerize.py [hunk 2]
```diff
@@ -57,38 +55,47 @@ def generate_swagger(package):
...
+                        if not params.get('body'):
+                            params['body'] = dict()
                         params['body'][name] = arg
 
                 parameters = []
                 # add all path params
-                for name, arg in six.iteritems(params['path']):
-                    parameters.append(_generate_param_dict(route, name, arg, 'path', True))
+                if params.get('path'):
+                    for name, arg in six.iteritems(params['path']):
```
Comment (posener) [inline hammock/swaggerize.py]: instead the if, use `six.iteritems(params.get('path', {}))`.
Same thing is few lines below

File: hammock/swaggerize.py [hunk 3]
```diff
@@ -142,13 +152,26 @@ def _join_hierarchy(parents, resource_class, route_method, sep):
     return sep.join(command)
 
 
+def represent_ordereddict(dumper, data):
```
Comment (posener) [inline hammock/swaggerize.py]: ouch :-)

File: hammock/swaggerize.py [hunk 4]
```diff
@@ -114,6 +122,8 @@ def generate_swagger(package):
                         }
                     }
                 }
+                if not paths.get(path):
```
Comment (posener) [inline hammock/swaggerize.py]: paths.setdefault(path, dict)

**Comment by ravidbro** [comment]
Fixed

File: hammock/swaggerize.py [hunk 5]
```diff
@@ -142,13 +141,26 @@ def _join_hierarchy(parents, resource_class, route_method, sep):
...
+    return yaml.nodes.MappingNode(u'tag:yaml.org,2002:map', value)
+
+
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('package')
     args = parser.parse_args()
     parser.add_mutually_exclusive_group()
     doc = generate_swagger(args.package)
-    json.dump(doc, fp=sys.stdout, indent=2)
```
Comment (shay-stratoscale) [inline hammock/swaggerize.py]: I think you should be backward compatible, and add a format parameter (default json)

Comment (ravidbro) [reply hammock/swaggerize.py]: like @rom-stratoscale said, I don't think so.
we want to be aligned to yaml in all services.
when someone will update hammock hash in his service, he will also commit a new swagger

**Pull request merged**